pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
But the reality is that `equals` only looks at the content (using `array_equivalent`) and not at the the object type, data type or attributes:

```
In [8]: pd.Index([1, 2]).equals([1, 2])
Out[8]: True

In [9]: pd.Index([1, 2]).equals(pd.Index([1, 2], name='name'))
Out[9]: True

In [10]: pd.Index([1, 2]).equals(pd.Index([1.0, 2.0]))
Out[10]: True

In [15]: pd.Index(['a', 'b']).equals(pd.Index(['a', 'b'], name='name'))
Out[15]: True
```

This also seems consistent with its docstring ("contain the same elements"). 
For the stricter comparison (also checking attributes), there is `identical`